### ISSUE TITLE
model linking graphs

### ISSUE THREAD

**Issue by whoburg** [hoburg/gpkit#814]
Here's a model linking graph from @pgkirsch's master's thesis:

![screen shot 2016-08-15 at 10 40 31 am](https://cloud.githubusercontent.com/assets/6042395/17667852/b759d536-62d4-11e6-99a8-3651a7265f16.png)

The code to produce this is below. Let's generalize it so it works for any `ConstraintSet`.

``` python
from math import sin, cos, pi
from gpkit.models.gpkitmodels.aircraft.aircraft import Aircraft
from gpkit.models.gpkitmodels.aircraft.fuselage import Fuselage
from gpkit.models.gpkitmodels.aircraft.landing_gear import LandingGear
from gpkit.models.gpkitmodels.aircraft.vtail import VerticalTail
from gpkit.models.gpkitmodels.aircraft.htail import HorizontalTail
from gpkit.models.gpkitmodels.aircraft.wing import Wing
from itertools import combinations

def make_figure():
    ac = Aircraft()
    subsystems = [Fuselage(), LandingGear(), VerticalTail(), HorizontalTail(),
                  Wing()]

    keychain = {}
    for vk in ac.varkeys:
        model = vk.descr.get("models")
        if model == ['Aircraft'] and vk not in ac.substitutions:
            v = vk.str_without(['models'])
            keychain[v] = []
            for ss in subsystems:
                if v in ss.varkeys:
                    keychain[v].append(type(ss).__name__)
            if len(keychain[v]) <= 1:
                del keychain[v]

    # Get all combinations of sub-models
    modellist = [type(ss).__name__ for ss in subsystems]
    modellistcombo = []
    for n in range(2,6):
        modellistcombo += combinations(modellist, n)

    # Create a dictionary that has each combination (tuple) as a key
    zidane = {}
    for modelgroup in modellistcombo:
        zidane[modelgroup] = []

    # Fill this dictionary in with all the varkeys for which each combo applies
    for key in keychain:
        zidane[tuple(keychain[key])] += [key]

    # Get rid of any empty entries
    viera = {k: v for k, v in zidane.items() if v}

    with open('figs/linked_vars.tex', 'w') as outfile:

        outfile.write('\\begin{center}\n' +
                      '\\begin{tikzpicture}[thick]\n')

        # Create a circular ring of nodes, one for each model
        nodepos = {}
        i = 0
        I = len(modellist)
        R = 6
        for model in modellist:
            nodepos[model] = (R*sin(2*pi*i/I), R*cos(2*pi*i/I))
            outfile.write('\\node at {0}'.format(str(nodepos[model])) +
                          '[circle, minimum size=3cm, fill=blue!20]' +
                          '({0}){{{0}}};\n'.format(model))
            i = i + 1

        j = 0
        colours = ['red', 'blue', 'cyan ', 'magenta', 'brown', 'gray', 
                   'olive', 'orange']
        for key in viera:
            # Create a node for every group of variables
            vargroup = viera[key]
            varnodeposx = 0
            varnodeposy = 0
            for k in key:
                varnodeposx += nodepos[k][0]
                varnodeposy += nodepos[k][1]
            varnodeposx = varnodeposx/len(key)
            varnodeposy = varnodeposy/len(key)
            outfile.write('\\node at (%.2f,%.2f)' % (varnodeposx, varnodeposy) +
                          '[draw,rectangle, color=%s, align=left, fill=white]({'
                          % colours[j] +
                          ' '.join(vargroup).replace('\\','') + '}){$' +
                          '$\\\\$'.join(vargroup) + '$};\n')
            # Create edges from vargroups to models they appear in 
            for k in key:
                outfile.write('\\begin{pgfonlayer}{background}\n' +
                              '\\draw[color=%s] ({' % colours[j] +
                              ' '.join(vargroup).replace('\\','') +
                              '})--('+k+');\n' +
                              '\\end{pgfonlayer}\n')

            j += 1

        outfile.write('\\end{tikzpicture}\n' +
                      '\\end{center}')

if __name__ == "__main__":
    make_figure()
```

**Comment by mayork** [comment]
adressed in PR #915 , waiting to merge

**Comment by bqpd** [comment]
Closed in #915
